### ISSUE TITLE
Fix S6605 FP: Should not be applied to expressions used by EntityFramework

### ISSUE THREAD

**Issue by KuraiAndras** [SonarSource/sonar-dotnet#7286]
### Description

The analyzer flags code inside expressions which will cause EF queries to be uncompilable.

I understand that checking for usage specifically related to EF might be problematic from the analyzer's side, but it is a very common use case and manually suppressing the issue is problematic. The analyzer itself is very useful in all other settings so turning it off by default is not the best solution.

### Repro steps

```csharp
public class MyEntity
{
    public Guid Id { get; set; }
}

public class MyDbContext : DbContext
{
    public DbSet<MyEntity> MyEntities { get; init; } = default!;
}

public class MyService
{
    private readonly MyDbContext _dbContext;

    public MyService(MyDbContext dbContext) => _dbContext = dbContext;

    // This compiles but is flagged by S6605
    public async Task DeleteEntities(List<Guid> ids)
    {
        var entitiesToDelete = await _dbContext
            .MyEntities
            .Where(e => ids.Any(i => e.Id == i))
            .ToListAsync();

        _dbContext.RemoveRange(entitiesToDelete);

        await _dbContext.SaveChangesAsync();
    }

    // This does not compile but is not flagged by S6605
    public async Task DeleteEntities2(List<Guid> ids)
    {
        var entitiesToDelete = await _dbContext
            .MyEntities
            .Where(e => ids.Exists(i => e.Id == i))
            .ToListAsync();

        _dbContext.RemoveRange(entitiesToDelete);

        await _dbContext.SaveChangesAsync();
    }
}
```

### Expected behavior

Not all collection-specific methods should be used inside expressions used by EntityFramework

### Actual behavior

Warning is displayed

### Known workarounds

Suppress the warning

### Related information

Sonar analyzer version `9.1.0.70676`

This case might also apply to other collection-specific warnings

**Comment by sebastien-marichal** [comment]
Hello @KuraiAndras,

Thank you for your feedback.

This is confirmed as a false positive.

**Comment by KuraiAndras** [comment]
Nice work guys 🚀
